PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = ['hr.employee']
-    _inherit = ['hr.employee.public']
-    _inherit = ['res.users']
-    _inherit = ['resource.resource']

Total Changes: 15

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_skills/models/hr_employee.py
+++ b/addons/hr_skills/models/hr_employee.py
@@ -5,7 +5,7 @@ from odoo.tools import convert
 
 
 class HrEmployee(models.Model):
-    _inherit = ['hr.employee']
+    _inherit = 'hr.employee'
 
     resume_line_ids = fields.One2many('hr.resume.line', 'employee_id', string="Resume lines")
     employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills",

--- a/addons/hr_skills/models/hr_employee_public.py
+++ b/addons/hr_skills/models/hr_employee_public.py
@@ -5,7 +5,7 @@ from odoo import fields, models
 
 
 class HrEmployeePublic(models.Model):
-    _inherit = ['hr.employee.public']
+    _inherit = 'hr.employee.public'
 
     resume_line_ids = fields.One2many('hr.resume.line', 'employee_id', string="Resume lines")
     employee_skill_ids = fields.One2many('hr.employee.skill', 'employee_id', string="Skills",

--- a/addons/hr_skills/models/hr_employee_skill.py
+++ b/addons/hr_skills/models/hr_employee_skill.py
@@ -8,6 +8,7 @@ from collections import defaultdict
 
 
 class HrEmployeeSkill(models.Model):
+    _name = 'hr.employee.skill'
     _description = "Skill level for an employee"
     _order = "skill_type_id, skill_level_id"
     _rec_name = "skill_id"

--- a/addons/hr_skills/models/hr_employee_skill_log.py
+++ b/addons/hr_skills/models/hr_employee_skill_log.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrEmployeeSkillLog(models.Model):
+    _name = 'hr.employee.skill.log'
     _description = "Skills History"
     _rec_name = 'skill_id'
     _order = "employee_id,date"

--- a/addons/hr_skills/models/hr_resume_line.py
+++ b/addons/hr_skills/models/hr_resume_line.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrResumeLine(models.Model):
+    _name = 'hr.resume.line'
     _description = "Resume line of an employee"
     _order = "line_type_id, date_end desc, date_start desc"
 

--- a/addons/hr_skills/models/hr_resume_line_type.py
+++ b/addons/hr_skills/models/hr_resume_line_type.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrResumeLineType(models.Model):
+    _name = 'hr.resume.line.type'
     _description = "Type of a resume line"
     _order = "sequence"
 

--- a/addons/hr_skills/models/hr_skill.py
+++ b/addons/hr_skills/models/hr_skill.py
@@ -5,6 +5,7 @@ from odoo import api, fields, models
 
 
 class HrSkill(models.Model):
+    _name = 'hr.skill'
     _description = "Skill"
     _order = "sequence, name"
 

--- a/addons/hr_skills/models/hr_skill_level.py
+++ b/addons/hr_skills/models/hr_skill_level.py
@@ -5,6 +5,7 @@ from odoo import api, fields, models, _
 
 
 class HrSkillLevel(models.Model):
+    _name = 'hr.skill.level'
     _description = "Skill Level"
     _order = "level_progress desc"
 

--- a/addons/hr_skills/models/hr_skill_type.py
+++ b/addons/hr_skills/models/hr_skill_type.py
@@ -5,6 +5,7 @@ from odoo import fields, models
 
 
 class HrSkillType(models.Model):
+    _name = 'hr.skill.type'
     _description = "Skill Type"
     _order = "name"
 

--- a/addons/hr_skills/models/res_users.py
+++ b/addons/hr_skills/models/res_users.py
@@ -4,7 +4,7 @@ from odoo import fields, models
 
 
 class ResUsers(models.Model):
-    _inherit = ['res.users']
+    _inherit = 'res.users'
 
     resume_line_ids = fields.One2many(related='employee_id.resume_line_ids', readonly=False)
     employee_skill_ids = fields.One2many(related='employee_id.employee_skill_ids', readonly=False)

--- a/addons/hr_skills/models/resource_resource.py
+++ b/addons/hr_skills/models/resource_resource.py
@@ -4,6 +4,6 @@ from odoo import fields, models
 
 
 class ResourceResource(models.Model):
-    _inherit = ['resource.resource']
+    _inherit = 'resource.resource'
 
     employee_skill_ids = fields.One2many(related='employee_id.employee_skill_ids')
